mvlearn/embed/dcca.py

```diff
@@ -0,0 +1,349 @@
+    optimizer : torch.optim.RMSprop object
+        Optimizer used to train the networks.
+    threshold : float
+        Threshold difference between successive iteration losses to define
+        convergence and stop training.
+
+    Notes
+    -----
+    Deep Canonical Correlation Analysis is a method of finding highly
+    correlated subspaces for 2 views of data using nonlinear transformations
+    learned by deep networks. It can be thought of as using deep networks
+    to learn the best potentially nonlinear kernels for a variant of kernel
+    CCA.
+
+    The networks used for each view in DCCA consist of fully connected linear
+    layers with sigmoid activation functions.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from mvlearn.embed.dcca import DCCA
+    >>> view1 = np.exp(np.random.normal(size=(1000, 100)))
+    >>> view2 = np.random.normal(loc=2, size=(1000, 75))
+    >>> input_size1, input_size2 = 100, 75
+    >>> outdim_size = 2
+    >>> layer_sizes1 = [1024, 2]
+    >>> layer_sizes2 = [1024, 2]
+    >>> dcca = DCCA(input_size1, input_size2, outdim_size, layer_sizes1,
+                    layer_sizes2)
+    >>> outputs = dcca.fit_transform([view1, view2])
+    >>> print(outputs[0].shape)
+    (1000, 2)
+
+    References
+    ----------
+    .. [#1DCCA] Andrew, G., Arora, R., Bilmes, J., & Livescu, K. (2013,
+                February). Deep canonical correlation analysis. In
+                International conference on machine learning (pp. 1247-1255).
+    """
+
+    def __init__(
+            self, input_size1=None, input_size2=None, outdim_size=2,
+            layer_sizes1=None, layer_sizes2=None,
+            use_all_singular_values=False, device=torch.device('cpu'),
+            epoch_num=10, batch_size=800, learning_rate=1e-3, reg_par=1e-5,
+            print_train_log_info=False, threshold=1e-2
+            ):
+
+        super().__init__()
+        # check input_size1/2
+
+        self.input_size1 = input_size1
+        self.input_size2 = input_size2
+        self.outdim_size = outdim_size
+
+        if layer_sizes1 is None:
+            self.layer_sizes1 = [1000, outdim_size]
+        if layer_sizes2 is None:
+            self.layer_sizes2 = [1000, outdim_size]
+
+        self.use_all_singular_values = use_all_singular_values
+        self.device = device
+        self.epoch_num = epoch_num
+        self.batch_size = batch_size
+        self.learning_rate = learning_rate
+        self.reg_par = reg_par
+        self.print_train_log_info = print_train_log_info
+        self.threshold = threshold
+
+        self.deep_model = DeepCCA(layer_sizes1, layer_sizes2, input_size1,
+                                  input_size2, outdim_size,
+                                  use_all_singular_values, device=device)
+        self.linear_cca = linear_cca()
+
+        self.model = nn.DataParallel(self.deep_model)
+        self.model.to(device)
+        self.loss = self.deep_model.loss
```

Comment:
you should probably include the loss function as part of the class unless it's needed in other methods. That is, don't include it in utils but include it in either specific dcca_utils or dcca class itself.